z/cache.go

```diff
@@ -0,0 +1,13 @@
+package z
+
+import "time"
+
+// Cache modeled after github.com/dgraph-io/ristretto#Cache and includes only subset of required methods
+type Cache interface {
```

Comment:
I'd prefer the `Cache interface` to live with the consumer of the interface. Export `struct`, import `interface`

Reply:
Yes, although, in this case, resolvers must be pulled out into their package, which is possible, although it will result in considerable refactoring w/o many functional improvements. I looked into it.